pkg/api/v20240812preview/openshiftcluster_validatestatic.go

```diff
@@ -520,9 +520,13 @@ func (sv openShiftClusterStaticValidator) validatePlatformIdentities(oc *OpenShi
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "identity", "Cluster identity and platform workload identities require each other.")
 	}
 
-	if operatorRolePresent && len(oc.Identity.UserAssignedIdentities) != 1 {
+	if clusterIdentityPresent && len(oc.Identity.UserAssignedIdentities) != 1 {
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "identity", "The provided cluster identity is invalid; there should be exactly one.")
 	}
 
+	if operatorRolePresent && len(pwip.PlatformWorkloadIdentities) == 0 {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "properties.platformWorkloadIdentityProfile", "Platform workload identity profile cannot be empty.")
```

Comment:
```suggestion
		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "properties.platformWorkloadIdentityProfile.platformWorkloadIdentities", "The set of platform workload identities cannot be empty.")
```

Suggesting this change because the error message should point to the exact property that we're validating to make the message easily actionable.

Reply:
Updated the comment.